src/awkward/_v2/_connect/rdataframe/from_rdataframe.py

```diff
@@ -66,7 +66,7 @@ def _wrap_as_record_array(array):
     if form_str.startswith("{"):
         form = ak._v2.forms.from_json(form_str)
         list_depth = form.purelist_depth
-        if list_depth > 3:
+        if list_depth > 4:
             raise ak._v2._util.error(NotImplementedError)
```

Comment:
Why is there a limit on the list depth? Since these things get generated by recursive functions, I might have expected practical limits (memory use, etc.) but not an in-principle limit like this.

Anyway, the NotImplementedError would have to explain what happened, since I'd be very confused if I got this error and wouldn't think that it's the list depth that's at fault.

Reply:
done